### PR TITLE
httpsession: prep work for supporting fetches other than next

### PR DESCRIPTION
**Pull request by jkarneges**
This reworks the next link handling in `HttpSession` to make it reusable for other things in the future. It should not introduce any change in behavior. Some `if(state == Proxying)` conditionals are added in places where `state` is known/assumed to already be of that value.

### CODE REVIEW

File: src/handler/httpsession.cpp [hunk 1]
```diff
@@ -1232,114 +1242,121 @@ class HttpSession::Private : public QObject
 				return;
 			}
 
-			if(outReq->bytesAvailable() > 0)
+			if(state == Proxying)
```
Comment (jkarneges) [inline src/handler/httpsession.cpp:1245]: this section is simply an indentation (and assertion of the state value)

File: src/handler/httpsession.cpp [hunk 2]
```diff
@@ -1477,21 +1494,24 @@ private slots:
 		{
 			haveOutReqHeaders = true;
 
-			// apply ProxyContent filters of all channels
-			QStringList allFilters;
-			foreach(const Instruct::Channel &c, instruct.channels)
+			if(state == Proxying)
```
Comment (jkarneges) [inline src/handler/httpsession.cpp:1497]: another indentation

File: src/handler/httpsession.cpp [hunk 3]
```diff
@@ -1501,34 +1521,38 @@ private slots:
...
 
-		delete responseFilters;
-		responseFilters = 0;
-
-		delete outReq;
-		outReq = 0;
+		cleanupOutReq();
 
-		log_debug("httpsession: failed to retrieve next link");
+		if(state == Proxying)
```
Comment (jkarneges) [inline src/handler/httpsession.cpp:1526]: another indentation based on state value

**Review by deg4uss3r — APPROVED** [review]
The moves in here look good to me, and pretty easily traceable, thanks Justin!

**Pull request merged**